### ISSUE TITLE
Partition with micro-rtps

### ISSUE THREAD

**Issue by lmayencourt** [eProsima/micro-RTPS#1]
What it the right way to specify a partition name for a topic with micro-rtps ?

I have try to modify the "data_writer_profile.xml" like this:
```
<profiles>
  <publisher profile_name="partition_xrce_publisher_profile">
    <topic>
      <kind>WITH_KEY</kind>
      <name>Square</name>
      <dataType>ShapeType</dataType>
      <historyQos>
        <kind>KEEP_LAST</kind>
        <depth>5</depth>
      </historyQos>
      <durability>
        <kind>TRANSIENT_LOCAL</kind>
      </durability>
    </topic>
        <qos>
            <partition>
                <names>
                    <name>A</name>
                </names>
            </partition>
       </qos>
    </publisher>
</profiles>
```

The agent parse properly the xml file and the `attributes.qos.m_partition` variable contains the proper name partition. However the Shapedemo application can not see any data under the Square topic with partition A. 
The only way I found to make it work is to hard code a partition name inside the agent `DataWriter::init()` function...

What I'm doing wrong ?

**Comment by BorjaOuterelo** [comment]
Hi @lmayencourt.
We will have a look at it and will come back to you.

**Comment by BorjaOuterelo** [comment]
Hi @lmayencourt 

We looked at this issue and It was caused by a bug on [Fast-RTPS](https://github.com/eProsima/Fast-RTPS). We have just solved the issue on Fast-RTPS. This way if you compile your Micro-RTPS Agent against that new version of Fast-RTPS all should be fine.

This is the change. Merged on branch master.
https://github.com/eProsima/Fast-RTPS/commit/b1779b608c7b5b2dcb101728f4213c58bdde74ee

If you have any other issue don't hesitate in getting back to us.

Thanks.
